frontend/src/components/Calendar.vue

```diff
@@ -121,7 +125,7 @@ function handleEventClick(info) {
 
 async function handleEventDrop(info) {
   try {
-    const transaction = transactionsStore.getTransactionById(parseInt(info.event.id))
+    const transaction = await transactionsStore.fetchTransactionById(parseInt(info.event.id))
```

Comment:
we might consider using get instead of fetch here to avoid the re-draw issue, but data could be stagnated